src/views/ProviderView/index.js

```diff
@@ -498,7 +498,7 @@ module.exports = class ProviderView {
       }
 
       // split url because chrome adds '#' to redirects
-      if (authWindowUrl && authWindowUrl.split('#')[0] === redirect) {
+      if (authWindowUrl && (authWindowUrl === redirect || authWindowUrl.split('#')[0] === redirect)) {
```

Comment:
Hi, what scenario does this solve? Testing in chrome, the current implementation already covers for both scenarios. 

```js
"yahoo.com?redherring=3894".split('#')
["yahoo.com?redherring=3894"]
"yahoo.com?redherring=3894#".split('#')
(2) ["yahoo.com?redherring=3894", ""]
```

Or does `.split` behave differently in other browsers?

Reply:
As I've mentioned in the description, this covers the case when the **redirect** contains a hash. 

For example, if `location.href = 'http://localhost/#/foo/bar'`, then `redirect` [gets assigned](https://github.com/transloadit/uppy/pull/828/files#diff-1018535a09fc115f567d294f012f2d2aR482): `http://localhost/#/foo/bar?id=12345`. Which makes `authWindowUrl` become `http://localhost/#/foo/bar?id=12345` as well.

So this `if` condition: `authWindowUrl.split('#')[0] === redirect` was failing (because `http://localhost/` != `http://localhost/#/foo/bar`).

But now that I think about it more, I think there's a bug on [line 482](https://github.com/transloadit/uppy/blob/a8c898354980ed75de4b53cbff37584720156ddd/src/views/ProviderView/index.js#L482) as well, because if the current location is: `http://localhost/#/foo/bar?x=z`, the `redirect` gets assigned: `http://example.com/#/foo/bar?x=z?id=12345`, which seems wrong.